src/state/slices/validatorDataSlice/validatorDataSlice.test.ts

```diff
@@ -0,0 +1,104 @@
+import { clearState, store } from 'state/store'
+
+import { selectSingleValidator } from './selectors'
+import { validatorData } from './validatorDataSlice'
+
+const SHAPESHIFT_VALIDATOR_ADDRESS = 'cosmosvaloper199mlc7fr6ll5t54w7tts7f4s0cvnqgc59nmuxf'
+const SHAPESHIFT_OPPORTUNITY = {
+  address: SHAPESHIFT_VALIDATOR_ADDRESS,
+  moniker: 'Shapeshift DAO',
+  tokens: '111111',
+  apr: '0.24',
+  commission: '0.100000000000000000',
+}
+
+describe('validatorDataSlice', () => {
+  beforeEach(() => {
+    clearState()
+  })
+
+  it('returns uninitialized properties for initialState', async () => {
+    expect(store.getState().validatorData).toEqual({
+      byValidator: {},
+      validatorIds: [],
+    })
+  })
+
+  describe('reducers', () => {
+    describe('upsertValidatorData', () => {
+      it('updates or inserts validator data in state', async () => {
+        store.dispatch(
+          validatorData.actions.upsertValidatorData({
+            validators: [],
```

Comment:
maybe we should add a similar test for upsertValidatorData with non empty data

Reply:
Actually made it with non-empty data everywhere.
It doesn't make sense to test inserting with an empty array from a testing perspective: I only had it there as empty array since we're testing core logic that's not related to the emptiness or lack thereof, of `validators` property, but if going for non-empty arrays, might as well have it not empty everywhere!
Thanks for the review 🐐